src/graphs/Composite.jl

```diff
@@ -0,0 +1,60 @@
+module Composite
+
+
+using ExtractMacro
+using ..Common
+using ..Interface
+
+export GraphComposite
+
+
+import ..Interface: energy, delta_energy, neighbors, update_cache!
+                     
+                    # TODO?: allΔE, delta_energy_residual, update_cache_residual!
+
+
+
+struct GraphComposite{G<:AbstractGraph} <: SimpleGraph{Float64}
+    N::Int
+    graphs::Vector{G}
+end
+
+function GraphComposite(graphs::AbstractGraph...)
+    length(graphs) >= 2 || throw(ArgumentError("at least 2 base graphs needed"))
+    gs = collect(graphs)
+    return GraphComposite(graphs[1].N, collect(graphs))
+end
+
+function energy(X::GraphComposite, C::Config)
+    sum(energy(x, C) for x in X.graphs)::Float64
+end
+
+function delta_energy(X::GraphComposite, C::Config, move::Int)
+    sum(delta_energy(x, C, move) for x in X.graphs)::Float64
+end
+
+function neighbors(X::GraphComposite, i::Int)
+    union(neighbors.(X.graphs, i)...)
+end 
+
+function update_cache!(X::GraphComposite, C::Config, move::Int)
+    map(x->update_cache!(x, C, move), X.graphs)
```

Comment:
```suggestion
    foreach(x->update_cache!(x, C, move), X.graphs)
```

(Since we don't need the return value. Then the following `return` can be removed too.)